sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -2053,6 +2053,34 @@ class ParquetV2FilterSuite extends ParquetFilterSuite {
       }
     }
   }
+
+  test("SPARK-38041: Data filter push down with partition filter") {
+    val data = spark.createDataFrame((1 to 4).map(i => Tuple1(Option(i))))
+    withNestedDataFrame(data).foreach { case (inputDF, colName, resultFun) =>
```

Comment:
I think the goal of this test to to check if the partition filters are also pushed down. It doesn't matter if the data filter is on nested column or not, so no need to test this on different levels of nested columns.

Reply:
Agreed. I will remove this in next commit.